domains/transfer/Transfer.API/API/Shared/Migrations/20231123093303_AddTransferAgreementProposalsAndDeleteConnections.cs

```diff
@@ -0,0 +1,85 @@
+                    SenderCompanyTin = table.Column<string>(type: "text", nullable: false),
+                    SenderCompanyName = table.Column<string>(type: "text", nullable: false),
+                    CreatedAt = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false, defaultValueSql: "current_timestamp at time zone 'UTC'"),
+                    StartDate = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false),
+                    EndDate = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: true),
+                    ReceiverCompanyTin = table.Column<string>(type: "text", nullable: true)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_TransferAgreementProposals", x => x.Id);
+                });
+        }
+
+        /// <inheritdoc />
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropTable(
+                name: "TransferAgreementProposals");
+
+            migrationBuilder.CreateTable(
+                name: "ConnectionInvitations",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    CreatedAt = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false, defaultValueSql: "current_timestamp at time zone 'UTC'"),
+                    SenderCompanyId = table.Column<Guid>(type: "uuid", nullable: false),
+                    SenderCompanyTin = table.Column<string>(type: "text", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_ConnectionInvitations", x => x.Id);
+                });
+
+            migrationBuilder.CreateTable(
+                name: "Connections",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    CompanyAId = table.Column<Guid>(type: "uuid", nullable: false),
+                    CompanyATin = table.Column<string>(type: "text", nullable: false),
+                    CompanyBId = table.Column<Guid>(type: "uuid", nullable: false),
+                    CompanyBTin = table.Column<string>(type: "text", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Connections", x => x.Id);
+                });
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Connections_CompanyAId",
+                table: "Connections",
+                column: "CompanyAId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Connections_CompanyBId",
+                table: "Connections",
+                column: "CompanyBId");
+        }
+    }
+}
```

Comment:
There should be a migration test for each migration. 

Setup the database state before the migration, then apply the migration, and assert if the new database state is correct, in the test. Take a look at the last migration test that was written in the certificates subsystem

Reply:
I know how a migration test works, thank you. I did the example if you remember

Reply:
But cool, I'll write one